File: lista-2/questao-1/src/dados/Filme.java
package dados;

class Filme {
    private String nome;
    private String lancamento;
    private String duracao;
    private Ator[] elenco = new Ator[50];
    private int quantElenco = 0;
    private Categoria categoria = new Categoria();

    public Filme() {

    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getNome() {
        return nome;
    }

    public void setDuracao(String duracao) {
        this.duracao = duracao;
    }

    public String getDuracao() {
        return duracao;
    }

    public void setLancamento(String lancamento) {
        this.lancamento = lancamento;
    }

    public String getLancamento() {
        return lancamento;
    }

    public void setElenco(Ator ator) {
        elenco[quantElenco] = ator;
        quantElenco++;
    }

    public Ator[] getElenco() {
        return elenco;
    }

    public void setCategoria(Categoria categoria) {
        this.categoria = categoria;
    }

    public Categoria getCategoria() {
        return categoria;
    }

    @Override
    public boolean equals(Object obj) {
        Filme u = new Filme();

        return u.equals(obj);
    }

    @Override
    public String toString() {
        StringBuffer string = new StringBuffer();
        string.append("Filme: \n");
        string.append(
                "\tNome: " + nome + "Lançamento: " + lancamento + "Duração: " + duracao + "Categoria: " + categoria);
        return string.toString();
    }
}

File: lista-2/trab/Questao-extra/Filme.py
from Ator import Ator


class Filme():

    def __init__(self, nome=0, lancamento=0, duracao=0):
        self.nome = nome
        self.lancamento = lancamento
        self.duracao = duracao
        self.quantElenco = 0
        self.elenco = []

    def setNome(self, nome):
        self.nome = nome

    def getNome(self):
        return self.nome

    def setLancamento(self, lancamento):
        self.lancamento = lancamento

    def getLancamento(self):
        return self.lancamento

    def setDuracao(self, duracao):
        self.duracao = duracao

    def getDuracao(self):
        return self.duracao

    def setElenco(self, ator):
        self.elenco.append(ator)
        self.quantElenco += 1

    def getElenco(self):
        return self.elenco

    def __str__(self):
        string = "Nome: {} | Lançamento: {} | Duração: {}".format(
            self.nome, self.lancamento, self.duracao)
        return string
